### PR TITLE
refactor(semver): make `increment()` more readable

### PR DESCRIPTION
**Pull request by timreichen**

### CODE REVIEW

File: semver/increment.ts [hunk 1]
```diff
@@ -73,153 +64,110 @@ export function increment(
...
-        };
-        break;
-      }
-    // 1.0.0 "pre" would become 1.0.0-0
-    // 1.0.0-0 would become 1.0.0-1
-    // 1.0.0-beta.0 would be come 1.0.0-beta.1
-    // switching the pre identifier resets the number to 0
-    case "pre":
-      result = {
+      const isPrerelease = (version.prerelease ?? []).length === 0;
```
Comment (iuioiua) [inline semver/increment.ts:147]:
```suggestion
      const isPrerelease = version.prerelease?.length === 0;
```

File: semver/increment.ts [hunk 2]
```diff
@@ -73,153 +64,110 @@ export function increment(
...
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    case "major":
-      // If this is a pre-major version, bump up to the same major version.
-      // Otherwise increment major.
+    case "prerelease": {
+      // If the input is a non-prerelease version, this acts the same as prepatch.
+      const isPrerelease = (version.prerelease ?? []).length === 0;
```
Comment (iuioiua) [inline semver/increment.ts:100]:
```suggestion
      const isPrerelease = version.prerelease?.length === 0;
```

File: semver/increment.ts [hunk 3]
```diff
@@ -73,153 +64,110 @@ export function increment(
...
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    case "minor":
-      // If this is a pre-minor version, bump up to the same minor version.
-      // Otherwise increment minor.
+      const isPrerelease = (version.prerelease ?? []).length === 0;
```
Comment (iuioiua) [inline semver/increment.ts:114]:
```suggestion
      const isPrerelease = version.prerelease?.length === 0;
```

File: semver/increment.ts [hunk 4]
```diff
@@ -73,153 +64,110 @@ export function increment(
...
-          major: version.major,
-          minor: version.minor,
-          patch: 0,
-          prerelease: [],
-          build: parseBuild(version.build, build),
-        };
-        break;
-      }
-    case "patch":
+      const isPrerelease = (version.prerelease ?? []).length === 0;
```
Comment (iuioiua) [inline semver/increment.ts:130]:
```suggestion
      const isPrerelease = version.prerelease?.length === 0;
```

**Review by iuioiua — APPROVED** [review]
LGTM. Nice!

File: semver/increment.ts (hunk 1 above)
Comment (timreichen) [reply semver/increment.ts]: That is not the same statement. 
`version.prerelease?.length === 0` returns false if `version.prerelease` is `undefined`
but
`(version.prerelease ?? []).length === 0` returns true if `version.prerelease`  is `undefined`

Comment (iuioiua) [reply semver/increment.ts]: Yep. Sorry. Will revert.

Comment (timreichen) [reply semver/increment.ts]: I guess one could write it as `!version.prerelease || version.prerelease.length === 0`

**Pull request merged**